### ISSUE TITLE
有没有办法只获得某一个变量的梯度，但是不进行optimizer.step()操作呢

### ISSUE THREAD

**Issue by OleNet** [PaddlePaddle/Paddle#21965]
有没有办法只获得某一个变量的梯度，类似于model.forward(); loss.backward(); 得到 model.input.gradient 但是不进行optimizer.step()操作呢

**Comment by iclementine** [comment]
静态图或者动态图？

如果是动态图的话，我们现在可以对一个 Variable 对象使用 grdient() 方法。这个方法在 loss.backward 之后， optimizer.minimize() 和 mode.clear_gradients() 之前使用。会返回 numpy array。
https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/fluid/framework.py#L1077

如果是静态图，你可以在一个 Variable 的名字后面加上 `@GRAD`，然后把这个名字添加进 fetch_list 里面去把它 fetch 出来查看。以及你也可以使用 gradients
https://www.paddlepaddle.org.cn/documentation/docs/zh/api_cn/backward_cn/gradients_cn.html#gradients 来计算梯度。
